internal/internal_task_pollers.go

```diff
@@ -365,10 +372,10 @@ func (wtp *workflowTaskPoller) processWorkflowTask(task *workflowTask) error {
 		if completedRequest == nil && err == nil {
 			return nil
 		}
-		if _, ok := err.(decisionHeartbeatError); ok {
+		if errors.As(err, new(*decisionHeartbeatError)) {
```

Comment:
yea, looks good - it's only returned from one place right now, and no wrapping, so this won't change behavior 👍

Reply:
This was not working before at all.
The error was returned as &decicisionHeartbeatError so type assertion err.(decisionHeartbeatError) was always false.

Reply:
ah, good point.  and good fix 👍